webpay/pay/tests/test_tasks.py

```diff
@@ -954,6 +954,17 @@ def test_prevent_reconfiguring_transaction(self):
         eq_(configured, False)  # Second call should do nothing.
         eq_(self.start_pay.call_count, 1)
 
+    def test_always_reconfigure_transaction_after_error(self):
```

Comment:
Could there be a test that shows that `session['configured_trans']` is only set when the connection succeeds?